### PR TITLE
organized the code / relocated packages for cleaner architecture

### PR DESCRIPTION
**Pull request by lpuskas**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | no
| API breaks?     | no
| Deprecations?   | no
| License         | Apache 2.0

### What's in this PR?
The code has been reorganized to meet the standard in our other projects (especially the usage of logging and error handling abstractions)(

### Why?
The code got easier to maintain

### Additional context
the functionality hasn't been changed, minor improvement s have been added though

### Checklist

- [X] Error handling code meets the [guideline](https://github.com/banzaicloud/pipeline/blob/master/docs/error-handling-guide.md)
- [X] Logging code meets the guideline (TODO)

### To Do
- testing (eg. : alpha)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -24,17 +24,23 @@ endif
...
-GOLANGCI_VERSION = 1.18.0
 MISSPELL_VERSION = 0.3.4
-JQ_VERSION = 1.5
-LICENSEI_VERSION = 0.1.0
-GOBIN_VERSION = 0.0.13
 GQLGEN_VERSION = 0.8.3
 PACKR_VERSION = 2.2.0
 
+GOLANGCI_VERSION = 1.20.0
+JQ_VERSION = 1.5
```
Comment (sagikazarmark) [inline Makefile]:
```suggestion
```

Not used

File: Makefile [hunk 2]
```diff
@@ -24,17 +24,23 @@ endif
...
+LICENSEI_VERSION = 0.2.0
+OPENAPI_GENERATOR_VERSION = v4.1.3
+MIGRATE_VERSION = 4.0.2
```
Comment (sagikazarmark) [inline Makefile]:
```suggestion
```

Not used

File: internal/app/cloudinfo/management/handler.go [hunk 3]
```diff
@@ -18,19 +18,19 @@ import (
...
 
+	"emperror.dev/emperror"
 	"github.com/gin-gonic/gin"
-	"github.com/goph/emperror"
 	"github.com/goph/logur"
 	"github.com/mitchellh/mapstructure"
 
 	"github.com/banzaicloud/cloudinfo/internal/app/cloudinfo/api"
-	"github.com/banzaicloud/cloudinfo/pkg/cloudinfo"
+	cloudinfo2 "github.com/banzaicloud/cloudinfo/internal/cloudinfo"
```
Comment (sagikazarmark) [inline internal/app/cloudinfo/management/handler.go]: IDE messed up the import

File: internal/cloudinfo/providers/alibaba/requests.go [hunk 4]
```diff
@@ -21,7 +21,7 @@ import (
 	"github.com/aliyun/alibaba-cloud-sdk-go/sdk/requests"
 	"github.com/aliyun/alibaba-cloud-sdk-go/sdk/responses"
 
-	"github.com/banzaicloud/cloudinfo/pkg/cloudinfo"
+	cloudinfo2 "github.com/banzaicloud/cloudinfo/internal/cloudinfo"
```
Comment (sagikazarmark) [inline internal/cloudinfo/providers/alibaba/requests.go]: IDE messed up the import

File: internal/cloudinfo/providers/amazon/cloudinfo_test.go [hunk 5]
```diff
@@ -25,9 +25,12 @@ import (
 	"github.com/goph/logur"
 	"github.com/stretchr/testify/assert"
 
-	"github.com/banzaicloud/cloudinfo/pkg/cloudinfo"
+	"github.com/banzaicloud/cloudinfo/internal/cloudinfo/cloudinfoadapter"
+	"github.com/banzaicloud/cloudinfo/internal/cloudinfo/types"
 )
 
+var logger = cloudinfoadapter.NewLogger(logur.NewTestLogger())
```
Comment (sagikazarmark) [inline internal/cloudinfo/providers/amazon/cloudinfo_test.go]: hm?

File: internal/cloudinfo/providers/amazon/network_mapper.go [hunk 6]
```diff
@@ -15,9 +15,10 @@
 package amazon
 
 import (
-	"github.com/pkg/errors"
+	"emperror.dev/errors"
 
-	"github.com/banzaicloud/cloudinfo/pkg/cloudinfo"
+	cloudinfo2 "github.com/banzaicloud/cloudinfo/internal/cloudinfo"
```
Comment (sagikazarmark) [inline internal/cloudinfo/providers/amazon/network_mapper.go]: IDE messed up the import

File: internal/cloudinfo/providers/azure/cloudinfo_test.go [hunk 7]
```diff
@@ -18,18 +18,21 @@ import (
...
 	"github.com/goph/logur"
-	"github.com/pkg/errors"
 	"github.com/stretchr/testify/assert"
 
-	"github.com/banzaicloud/cloudinfo/pkg/cloudinfo"
+	"github.com/banzaicloud/cloudinfo/internal/cloudinfo/cloudinfoadapter"
+	"github.com/banzaicloud/cloudinfo/internal/cloudinfo/types"
 )
 
+var cloudinfoLogger = cloudinfoadapter.NewLogger(logur.NewTestLogger())
```
Comment (sagikazarmark) [inline internal/cloudinfo/providers/azure/cloudinfo_test.go]: global

File: internal/cloudinfo/providers/oracle/network_mapper.go [hunk 8]
```diff
@@ -15,17 +15,18 @@
 package oracle
 
 import (
-	"github.com/pkg/errors"
+	"emperror.dev/errors"
 
-	"github.com/banzaicloud/cloudinfo/pkg/cloudinfo"
+	cloudinfo2 "github.com/banzaicloud/cloudinfo/internal/cloudinfo"
```
Comment (sagikazarmark) [inline internal/cloudinfo/providers/oracle/network_mapper.go]: IDE messed up the import

File: internal/app/cloudinfo/management/handler.go (hunk 3 above)
Comment (lpuskas) [reply internal/app/cloudinfo/management/handler.go]: thx!

File: internal/cloudinfo/providers/alibaba/requests.go (hunk 4 above)
Comment (lpuskas) [reply internal/cloudinfo/providers/alibaba/requests.go]: thx

File: internal/cloudinfo/providers/amazon/network_mapper.go (hunk 6 above)
Comment (lpuskas) [reply internal/cloudinfo/providers/amazon/network_mapper.go]: thx

File: internal/cloudinfo/providers/azure/cloudinfo_test.go (hunk 7 above)
Comment (lpuskas) [reply internal/cloudinfo/providers/azure/cloudinfo_test.go]: fixed

File: internal/cloudinfo/providers/amazon/cloudinfo_test.go (hunk 5 above)
Comment (lpuskas) [reply internal/cloudinfo/providers/amazon/cloudinfo_test.go]: thx

**Pull request merged**